test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -801,67 +801,135 @@ Describe "Invoke-WebRequest tests" -Tags "Feature" {
+            $uri = Get-WebListenerUrl -Test 'Get'
+            $UserAgent = 'Invalid:Agent'
+            $response = ExecuteRequestWithCustomUserAgent -Uri $uri -UserAgent $UserAgent  -Cmdlet "Invoke-WebRequest"
 
-        $response.Error | Should Not BeNullOrEmpty
-        $response.Error.FullyQualifiedErrorId | Should Be "System.FormatException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand"
-        $response.Error.Exception.Message | Should Be "The format of value 'Invalid:Agent' is invalid."
-    }
+            $response.Error | Should Not BeNullOrEmpty
+            $response.Error.FullyQualifiedErrorId | Should Be "System.FormatException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand"
+            $response.Error.Exception.Message | Should Be "The format of value 'Invalid:Agent' is invalid."
+        }
 
-    It "Verifies Invoke-WebRequest UserAgent handling does not report an error when using -SkipHeaderValidation" {
-        $uri = Get-WebListenerUrl -Test 'Get'
-        $UserAgent = 'Invalid:Agent'
-        $response = ExecuteRequestWithCustomUserAgent -Uri $uri -UserAgent $UserAgent  -SkipHeaderValidation -Cmdlet "Invoke-WebRequest"
+        It "Verifies Invoke-WebRequest UserAgent handling does not report an error when using -SkipHeaderValidation" {
+            $uri = Get-WebListenerUrl -Test 'Get'
+            $UserAgent = 'Invalid:Agent'
+            $response = ExecuteRequestWithCustomUserAgent -Uri $uri -UserAgent $UserAgent  -SkipHeaderValidation -Cmdlet "Invoke-WebRequest"
 
-        $response.Error | Should BeNullOrEmpty
-        $Pattern = [regex]::Escape($UserAgent)
-        $response.Content.Headers."User-Agent" | Should Match $Pattern
-    }
+            $response.Error | Should BeNullOrEmpty
+            $Pattern = [regex]::Escape($UserAgent)
+            $response.Content.Headers."User-Agent" | Should Match $Pattern
+        }
+
+        It "Verifies Invoke-WebRequest default ContentType handling reports no error is returned for a valid Content-Type header value and -Body" {
+            $contentType = 'text/plain'
+            $body = "bar"
+            $uri = Get-WebListenerUrl -Test 'Post'
+
+            $response = Invoke-WebRequest -Uri $uri -Method 'Post' -ContentType $contentType -Body $body
+            $result = $response.Content | ConvertFrom-Json
+
+            $result.data | Should BeExactly $body
+            $result.headers.'Content-Type' | Should BeExactly $contentType
+        }
+
+        It "Verifies Invoke-WebRequest default ContentType handling reports an error is returned for an invalid Content-Type header value and -Body" {
+            $contentType = 'foo'
+            $body = "bar"
+            $uri = Get-WebListenerUrl -Test 'Post'
+
+            { Invoke-WebRequest -Uri $uri -Method 'Post' -ContentType $contentType -Body $body -ErrorAction 'Stop' } |
+                ShouldBeErrorId "WebCmdletContentTypeException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand"
```

Comment:
can we use the new Pester `should -throw -errorid <...>`?

Reply:
We could. I made this PR before the project changed to Pester 4. However, The guidance does still say to use `ShouldBeErrorId` https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/WritingPesterTests.md